bing-wallpapers.ps1

```diff
@@ -60,36 +77,60 @@ if (!(Test-Path $downloadFolder)) {
     New-Item -ItemType Directory $downloadFolder
 }
 
-$request = Invoke-WebRequest -Uri $uri
-[xml]$content = $request.Content
-
+# Add paging support for when number requested > 8
+# &idx=0&n=$maxItemCount
+$pageSize = 8
 $items = New-Object System.Collections.ArrayList
-foreach ($xmlImage in $content.images.image) {
-    [datetime]$imageDate = [datetime]::ParseExact($xmlImage.startdate, 'yyyyMMdd', $null)
-    [string]$imageUrl = "$hostname$($xmlImage.urlBase)_$resolution.jpg"
-
-    # Add item to our array list
-    $item = New-Object System.Object
-    $item | Add-Member -Type NoteProperty -Name date -Value $imageDate
-    $item | Add-Member -Type NoteProperty -Name url -Value $imageUrl
-    $null = $items.Add($item)
-}
-
-# Keep only the most recent $files items to download
-if (!($files -eq 0) -and ($items.Count -gt $files)) {
-    # We have too many matches, keep only the most recent
-    $items = $items|Sort date
-    while ($items.Count -gt $files) {
-        # Pop the oldest item of the array
-        $null, $items = $items
+if ($files -gt 0) {
+    if ($useJsonSource) {
+        $jsonImgs = ConvertFrom-Json -InputObject $(Invoke-WebRequest -Uri 'https://api45gabs.azurewebsites.net/api/sample/bingphotos')

```

Comment:
If I’m not mistaken, this endpoint is not an official one. I’m therefore hesitant to include it in this script. If you can find an ‘official’ endpoint that we could use, that would be perfect!